### PR TITLE
Feature.fix gmu pathelement unnamed load 411

### PR DESCRIPTION
**Pull request by zancas**
@pjbreaux   major refactor

### CODE REVIEW

File: f5/bigip/resource.py [hunk 1]
```diff
@@ -160,9 +160,103 @@ def __init__(self, container):
...
-        base_uri = self.__class__.__name__.lower() + '/'
-        self._meta_data['uri'] =\
-            self._meta_data['container']._meta_data['uri'] + base_uri
+        self._set_meta_data_uri()
+
+    def _set_meta_data_uri(self):
+        base_uri = self.__class__.__name__.lower()
+        # Handle 'terminal s or _s'
+        if isinstance(self, Collection):
+            if self.__class__.__name__.lower()[-2:] == '_s':
```
Comment (pjbreaux) [inline f5/bigip/resource.py:169]: Could also use str.endswith

File: f5/bigip/tm/net/dns_resolver.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""BIG-IP® Network ARP module.
+
+REST URI
+    ``http://localhost/mgmt/tm/net/arp``
```
Comment (pjbreaux) [inline f5/bigip/tm/net/dns_resolver.py]: Update manually

File: f5/bigip/tm/net/dns_resolver.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        self._meta_data['attribute_registry'] = {
+            'tm:net:dns-resolver:dns-resolverstate': Dns_Resolver
+        }
+
+
+class Dns_Resolver(Resource):
+    """BIG-IP® Dns Resolver resource."""
+    def __init__(self, Dns_Resolvers):
+        super(Dns_Resolver, self).__init__(Dns_Resolvers)
+        self._meta_data['allowed_lazy_attributes'] = [Dns_Resolver]
```
Comment (pjbreaux) [inline f5/bigip/tm/net/dns_resolver.py]: Is this necessary?

File: test/functional/auth/test_password_policy.py [hunk 4]
```diff
@@ -13,17 +13,21 @@
...
+from pprint import pprint as pp
+
 
 class TestPasswordPolicy(object):
     def test_load(self, bigip):
-        pp = bigip.auth.password_policy.load()
-        assert pp.maxLoginFailures == 0
-        pp.refresh()
-        assert pp.maxLoginFailures == 0
+        password_policy = bigip.auth.password_policy.load()
```
Comment (pjbreaux) [inline test/functional/auth/test_password_policy.py:21]: Should this test be more inline with what @tar saw when we loaded an object twice?

**Comment by zancas** [comment]
Per @pjbreaux 's comment I've opened issue #415 .

**Pull request merged**